### PR TITLE
Update mibig downloader

### PR DESCRIPTION
**Pull request by CunliangGeng**
add creation of mibig dir if not exist

add the support of PathLike

### CODE REVIEW

**Comment by CunliangGeng** [comment]
* **#218** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/218?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#217** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/217?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#215** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/215?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#214** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/214?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#212** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/212?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#211** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/211?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#210** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/210?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#209** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/209?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#208** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/208?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `dev`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CunliangGeng and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/nplinker/genomics/mibig/mibig_downloader.py [hunk 1]
```diff
@@ -23,31 +24,37 @@
...
+        folders will be removed if they exist. So the `extract_path` will have only json files.
+
     Args:
-        download_root(str): Path to the directory to place downloaded archive in
-        extract_path(str): Path to the empty directory metadata json files will
-            be extracted to.
-            Note that if it's a directory containing the metadata json files
-            after extraction, the json files will be moved to `extract_path`,
-            and then the extracted directory will be removed.
+        download_root(str): Path to the directory to place downloaded archive in.
```
Comment (gcroci2) [inline src/nplinker/genomics/mibig/mibig_downloader.py]:
```suggestion
        download_root(str | os.PathLike): Path to the directory in which to place the downloaded archive.
```

File: src/nplinker/genomics/mibig/mibig_downloader.py [hunk 2]
```diff
@@ -23,31 +24,37 @@
...
+        extract_path(str): Path to an empty directory where json files will be extracted to.
```
Comment (gcroci2) [inline src/nplinker/genomics/mibig/mibig_downloader.py]:
```suggestion
        extract_path(str | os.PathLike): Path to an empty directory where the json files will be extracted.
```

File: src/nplinker/genomics/mibig/mibig_downloader.py [hunk 3]
```diff
@@ -23,31 +24,37 @@
...
+            The directory must be empty if it exists. If not exist, the directory will be created.
```
Comment (gcroci2) [inline src/nplinker/genomics/mibig/mibig_downloader.py]:
```suggestion
            The directory must be empty if it exists. If it doesn't exist, the directory will be created.
```

File: src/nplinker/genomics/mibig/mibig_downloader.py [hunk 4]
```diff
@@ -23,31 +24,37 @@
...
 def download_and_extract_mibig_metadata(
-    download_root: str,
-    extract_path: str,
+    download_root: str | os.PathLike,
+    extract_path: str | os.PathLike,
     version: str = "3.1",
 ):
     """Download and extract MIBiG metadata json files.
 
+        Note that no matter if metadata json files locate in nested folders or not in the archive,
```
Comment (gcroci2) [inline src/nplinker/genomics/mibig/mibig_downloader.py]:
```suggestion
        Note that it does not matter whether the metadata json files are in nested folders or not in the archive,
```

File: src/nplinker/genomics/mibig/mibig_downloader.py [hunk 5]
```diff
@@ -23,31 +24,37 @@
...
+        all of the json files will be extracted to the same position i.e. `extract_path`. The nested
```
Comment (gcroci2) [inline src/nplinker/genomics/mibig/mibig_downloader.py]:
```suggestion
        all json files will be extracted to the same location, i.e. `extract_path`. The nested
```

**Review by gcroci2 — APPROVED** [review]
Just minor suggestions for the doc string.

**Comment by CunliangGeng** [comment]
### Merge activity

* **Mar 1, 9:49 AM EST**: @CunliangGeng started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/210).
* **Mar 1, 9:52 AM EST**: [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/210) rebased this pull request as part of a merge.
* **Mar 1, 9:53 AM EST**: @CunliangGeng merged this pull request with [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/210).

**Pull request merged**